cmd/src/extensions_publish.go

```diff
@@ -87,12 +94,32 @@ Notes:
 			}
 		} else {
 			// Prepare and upload bundle.
-			if err := runManifestPrepublishScript(manifest, filepath.Dir(*manifestFlag)); err != nil {
+			if err := runManifestPrepublishScript(manifest, manifestDir); err != nil {
 				return err
 			}
 
 			var err error
-			bundle, sourceMap, err = readExtensionArtifacts(manifest, filepath.Dir(*manifestFlag))
+			bundle, sourceMap, err = readExtensionArtifacts(manifest, manifestDir)
+			if err != nil {
+				return err
+			}
+		}
+
+		if *gitHeadFlag != "" {
+			manifest, err = updatePropertyInManifest(manifest, "gitHead", *gitHeadFlag)
+			if err != nil {
+				return err
+			}
+		} else {
+			command := exec.Command("git", "rev-parse", "head")
+			command.Dir = manifestDir
+
+			out, err := command.CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("failed to determine git head: %q", err)
```

Comment:
Maybe we should ignore this error (not add the field in that case) so that it's possible to publish an extension without having a git repo (in a simple folder) or before making the initial commit

Reply:
If we rely on this for troubleshooting I'd rather make sure we have it or else require the user to explicitly ignore it on publish (they can supply a `-git-head 'canned string'` on the command line to do this).

Thoughts?

Reply:
Should we add your suggestion in this comment to the error message then?

Reply:
Here are some arguments for ignoring the error:

- It's how `npm publish` behaves
- It would be a breaking change
- We don't require the `repository` field in package.json either afaik, so it's a bit weird to require the commit, but not the repository
- I'm worried that it increases the barrier to create extensions
- If the goal is to have this information primarily for code intel and our own extensions, we can pretty much guarantee that they always will be in a repo and have that info
  - just not for customer's private extensions (for which the commit is not useful to us anyway)
- We could still print a warning

I don't care strongly about this though so feel free to pick either behavior and merge that.